GithubJobsEnterpriseProject/Controllers/GithubJobsController.cs

```diff
@@ -0,0 +1,228 @@
+            }
+
+            _context.Entry(githubJob).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!GithubJobExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return NoContent();
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<GithubJob>> PostGithubJob(GithubJob githubJob)
+        {
+            _context.JobItems.Add(githubJob);
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateException)
+            {
+                if (GithubJobExists(githubJob.Id))
+                {
+                    return Conflict();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return CreatedAtAction("GetGithubJob", new { id = githubJob.Id }, githubJob);
+        }
+
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> DeleteGithubJob(string id)
+        {
+            var githubJob = await _context.JobItems.FindAsync(id);
+            if (githubJob == null)
+            {
+                return NotFound();
+            }
+
+            _context.JobItems.Remove(githubJob);
+            await _context.SaveChangesAsync();
+
+            return NoContent();
+        }
+
+        private bool GithubJobExists(string id)
+        {
+            return _context.JobItems.Any(e => e.Id == id);
+        }
+
+        [HttpGet("/register/username={username}&email={email}&password={password}")]
+        public bool GetVerify(string username, string email, string password)
+        {
+            foreach (var user in _userContext.Users)
+            {
+                if (user.Email == email || user.Username == username)
+                {
+                    return false;
+                }
+            }
+            Save(username,email,password);
+            return true;
+
+        }
+
+        [HttpGet("/getCookieData")]
+        public string GetCookieData()
+        {
+            var user = HttpContext.User;
+            return user.Identity.Name;
+        }
+
+        public void Save(string username, string email, string password)
+        {
+            var hashedPassword = _hashService.Hash(password);
+            _userContext.Users.Add(new User(username, email, hashedPassword));
+            _userContext.SaveChanges();
+        }
+
+        [HttpGet("/login/username={username}&password={password}")]
+        public bool Login(string username, string password)
+        {
+            var users = _userContext.Users.ToList();
+            var loginService = new LoginService(username, password, users);
```

Comment:
Avoid instantiation. `New is GLUE!!!`. Dependency injection should be applied here as well!